src/System.Net.Http/tests/UnitTests/HPack/HPackRoundtripTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http.HPack;
+using System.Net.Http.Headers;
+using System.Text;
+using Xunit;
+
+namespace System.Net.Http.Unit.Tests.HPack
+{
+    public class HPackRoundtripTests
+    {
+
+        public static IEnumerable<object[]> TestHeaders =>
+            new HttpHeaders[]
+            {
+                BuildHttpHeaders(("header", new[] { "value" })),
+
+                BuildHttpHeaders(("header", new[] { "value1", "value2" })),
+
+                BuildHttpHeaders(
+                    ("header-0", new[] { "value1", "value2" }),
+                    ("header-0", new[] { "value3" }),
+                    ("header-1", new[] { "value1" }),
+                    ("header-2", new[] { "value1", "value2" }))
+
+            }.Select(h => new[] { h });
```

Comment:
All this use of LINQ and the extra BuildHttpHeaders helper could be avoided with just:
```C#
public static IEnumerable<object[]> TestHeaders()
{
    yield return new object[] { new HttpRequestHeaders() { { "header", "value" } } };
    yield return new object[] { new HttpRequestHeaders() { { "header", new[] { "value1", "value2" } } } };
    yield return new object[] { new HttpRequestHeaders()
    {
        { "header-0", new[] { "value1", "value2" } },
        { "header-0", "value3" },
        { "header-1", "value1" },
        { "header-2", new[] { "value1", "value2" } },
    } };
}
```

Reply:
Done.